src/transformers/trainer.py

```diff
@@ -847,6 +847,15 @@ def create_optimizer(self):
                 )
             else:
                 self.optimizer = optimizer_cls(optimizer_grouped_parameters, **optimizer_kwargs)
+                if optimizer_cls.__name__ == "Adam8bit":
+                    import bitsandbytes
+
+                    manager = bitsandbytes.optim.GlobalOptimManager.get_instance()
+
+                    for module in self.model.modules():
+                        if isinstance(module, nn.Embedding):
+                            manager.register_module_override(module, "weight", {"optim_bits": 32})
+                            logger.info(f"Registering bitsandbytes override for {module}")
```

Comment:
Let's have something more user-comprehensible here? And perhaps switch it to `debug` as this is a way too low level of info and most users won't care about - since they can't change anything anyway.

Perhaps:

```suggestion
                            logger.debug(f"bitsandbytes: will optimize {module} in fp32")
```

An informative log would be a single - "using bitsandbytes AdamW optimizer with fp32 for embeddings, int8 for the rest of weights"?